instagram/src/components/CommentSection/CommentSection.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+
+import Comment from './Comment';
+import CommentInput from './CommentInput';
+import PropTypes from 'prop-types';
+
+class CommentSection extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            comments: props.comments,
+            comment: ''
+        }
+    }
+
+    
+    componentDidMount() {
+        let id = this.props.postId;
+        
+        if(localStorage.getItem(id)) {
+            this.setState({ comments: JSON.parse(localStorage.getItem(this.props.postId)) });
+        } else {
+            this.saveToLocalStorage();
+        }
+    }
+
+    componentWillUnmount() {
+        this.saveToLocalStorage();
+    }
+
+    saveToLocalStorage = _ => {
+        localStorage.setItem(this.props.postId, JSON.stringify(this.state.comments));
+    }
+
+    addNewComment = e => {
+        e.preventDefault();
+        let newComment = {text: this.state.comment, username: 'jonathan'};
```

Comment:
now that you've got login, think about how you would go about making this username the actual name that they used while signing in